daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md

```diff
@@ -0,0 +1,74 @@
+---
+type: docs
+title: "CockroachDB"
+linkTitle: "CockroachDB"
+description: Detailed information on the CockroachDB state store component
+aliases:
+  - "/operations/components/setup-state-store/supported-state-stores/setup-cockroachdb/"
+---
+
+## Create a Dapr component
+
+Create a file called `cockroachdb.yaml`, paste the following and replace the `<CONNECTION STRING>` value with your connection string. The connection string for CockroachDB follow the same standard for PostgreSQL connection string. For example, `"host=localhost user=root port=26257 connect_timeout=10 database=dapr_test"`. See the CockroachDB [documentation on database connections](https://www.cockroachlabs.com/docs/stable/connect-to-the-database.html) for information on how to define a connection string.
+
+If you want to also configure CockroachDB to store actors, add the `actorStateStore` configuration element shown below.
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: <NAME>
+  namespace: <NAMESPACE>
+spec:
+  type: state.cockroachdb
+  version: v1
+  metadata:
+  - name: connectionString
+    value: "<CONNECTION STRING>"
+```
+
+## Spec metadata fields
+
+| Field              | Required | Details | Example |
+|--------------------|:--------:|---------|---------|
+| connectionString   | Y        | The connection string for CockroachDB | `"host=localhost user=root port=26257 connect_timeout=10 database=dapr_test"`
+| actorStateStore    | N        | Consider this state store for actors. Defaults to `"false"` | `"true"`, `"false"`
+
+
+If you wish to use CockroachDB as an actor store, append the following to the yaml.
+
+```yaml
+  - name: actorStateStore
+    value: "true"
+```
+
+
+## Create CockroachDB
+
+{{< tabs "Self-Hosted" >}}
```

Comment:
Can you also provide an example of how to deploy to Kubernetes? Also any other hosting environments if these are easy to reference into the CockroachDB docs (e.g. Deploying to production VM environment)

Reply:
Of course, I'll make the adjustment 😃